src/app/pages/data-protection/snapshot/snapshot-form/snapshot-form.component.ts

```diff
@@ -1,244 +1,151 @@
-          tooltip: helptext.begin_tooltip,
-          options: [],
-          value: '00:00',
-          required: true,
-          validation: [Validators.required],
-        }, {
-          type: 'select',
-          name: 'end',
-          placeholder: helptext.end_placeholder,
-          tooltip: helptext.end_tooltip,
-          options: [],
-          value: '23:59',
-          required: true,
-          validation: [Validators.required],
-        }, {
-          type: 'checkbox',
-          name: 'allow_empty',
-          placeholder: helptext.allow_empty_placeholder,
-          tooltip: helptext.allow_empty_tooltip,
-          value: true,
-        }, {
-          type: 'checkbox',
-          name: 'enabled',
-          placeholder: helptext.enabled_placeholder,
-          tooltip: helptext.enabled_tooltip,
-          value: true,
-        },
-      ],
-    },
-    { name: 'divider', divider: true },
-  ]);
+  get title(): string {
+    return this.isNew
+      ? this.translate.instant('Add Periodic Snapshot Task')
+      : this.translate.instant('Edit Periodic Snapshot Task');
+  }
 
-  constructor(protected taskService: TaskService,
+  form = this.fb.group({
+    dataset: ['', Validators.required],
+    exclude: [[] as string[]],
+    recursive: [false],
+    lifetime_value: [2, Validators.required],
+    lifetime_unit: [LifetimeUnit.Week, Validators.required],
+    naming_schema: ['auto-%Y-%m-%d_%H-%M', [Validators.required, Validators.pattern('[^/]+')]],
+    schedule: [CronPresetValue.Daily as string, Validators.required],
+    begin: ['00:00', Validators.required],
+    end: ['23:59', Validators.required],
+    allow_empty: [true],
+    enabled: [true],
+  });
+
+  isLoading = false;
+
+  readonly labels = {
+    dataset: helptext.dataset_placeholder,
+    exclude: helptext.exclude_placeholder,
+    recursive: helptext.recursive_placeholder,
+    lifetime: helptext.lifetime_placeholder,
+    naming_schema: helptext.naming_schema_placeholder,
+    schedule: helptext.snapshot_picker_placeholder,
+    begin: helptext.begin_placeholder,
+    end: helptext.end_placeholder,
+    allow_empty: helptext.allow_empty_placeholder,
+    enabled: helptext.enabled_placeholder,
+  };
+
+  readonly tooltips = {
+    dataset: helptext.dataset_tooltip,
+    exclude: helptext.exclude_tooltip,
+    recursive: helptext.recursive_tooltip,
+    lifetime: helptext.lifetime_tooltip,
+    naming_schema: helptext.naming_schema_tooltip,
+    schedule: helptext.snapshot_picker_tooltip,
+    begin: helptext.begin_tooltip,
+    end: helptext.end_tooltip,
+    allow_empty: helptext.allow_empty_tooltip,
+    enabled: helptext.enabled_tooltip,
+  };
+
+  readonly datasetOptions$ = this.storageService.getDatasetNameOptions();
+  readonly timeOptions$ = of(this.taskService.getTimeOptions());
+  readonly lifetimeOptions$ = of(
+    Object.values(LifetimeUnit).map((lifetime) => ({ label: lifetime + '(S)', value: lifetime })),
```

Comment:
'(S)' will not work in other languages. I think it would be okay to keep lifetime units as is.